broadcast/src/buffered/engine.rs

```diff
@@ -277,31 +366,31 @@ impl<
             .or_insert_with(|| VecDeque::with_capacity(self.deque_size + 1));
 
         // If the message is already in the deque, move it to the front and return early
-        if let Some(i) = deque.iter().position(|d| *d == digest) {
+        if let Some(i) = deque.iter().position(|d| d.0 == identity && d.1 == digest) {
```

Comment:
I think a `type Pair { identity: Di, digest: Dd }` would make this code easier to read. For example you could derive `Eq` here to make it easier, and below you can avoid using `.0` and `.1`

Reply:
Agreed.